soccer/gameplay/plays/offense/distraction.py

```diff
@@ -0,0 +1,178 @@
+            #if the ball is already near the striker, then no pass will occur and the striker will just cpature the ball
+            self.add_subbehavior(skills.capture.Capture(), 'striker get close ball', required = True)
+        else:
+            if pass_to_distract_chance >= pass_to_striker_chance*shot_of_striker_chance1:
+                self.add_subbehavior(skills.move.Move(self.striker_point), 'make striker stay again', required = True)
+                self.add_subbehavior(tactics.coordinated_pass.CoordinatedPass(self.distraction_recieve_pass_point), 'distract pass', required = True)
+            else:
+                self.add_subbehavior(skills.move.Move(self.distraction_point), 'make striker stay again', required = True)
+                self.add_subbehavior(tactics.coordinated_pass.CoordinatedPass(self.striker_point), 'striker pass', required = True) 
+
+    def on_exit_passing(self):
+        self.remove_all_subbehaviors()
+
+    def on_enter_cross(self):
+        #if the ball is passed to the distractor the ball is passed to the striker, as the third robot moves to the right to distract more
+        pass_to_striker_chance = evaluation.passing.eval_pass(main.ball().pos, self.striker_point, main.our_robots() )
+        if self.striker_box_max.x < main.ball().pos.x < self.striker_box_min.x and self.striker_box_max.y > main.ball().pos.y > self.striker_box_min.y:
+            #if the ball is already near the striker, then no pass will occur and the striker will just cpature the ball
+            self.add_subbehavior(skills.capture.Capture(), 'capture 3', required = True)
+        #add chip here if pass chance is low
+        else:
+            self.add_subbehavior(tactics.coordinated_pass.CoordinatedPass(self.striker_point), 'pass to striker', required = True)
+            self.add_subbehavior(skills.move.Move(self.distraction_point), 'move distract', required = False, priority = 10)
+            self.add_subbehavior(skills.move.Move(self.distraction_recieve_pass_point), 'shift right', required = False, priority = 10)
+
+    def on_exit_cross(self):
+        self.remove_all_subbehaviors()
+
+    def on_enter_shoot(self):
+        #Depending on shot and pass chances,
+        #the striker will shoot
+        #or
+        #the striker will pass to the distractor and the distractor will shoot
+        pass_striker_to_distractor_chance = evaluation.passing.eval_pass( self.striker_point, self.distraction_recieve_pass_point, main.our_robots() )
+        shot_of_striker_chance = evaluation.shooting.eval_shot( self.striker_point, main.our_robots() )
+        shot_of_distractor_chance = evaluation.shooting.eval_shot( self.distraction_recieve_pass_point, main.our_robots() )
+
+        if pass_striker_to_distractor_chance*shot_of_distractor_chance > shot_of_striker_chance:
+                self.add_subbehavior(tactics.coordinated_pass.CoordinatedPass(self.distraction_recieve_pass_point), 'distract pass', required = True)
```

Comment:
As talked about, there should be a transition to either the pass or cross state instead of trying to do a pass in the shoot state